core/new-gui/src/app/workspace/service/workflow-websocket/workflow-websocket.service.ts

```diff
@@ -98,6 +101,44 @@ export class WorkflowWebsocketService {
     this.websocketEvent().subscribe(_ => (this.isConnected = true));
   }
 
+  public openExecutionWebsocket(eId: number, workflowGraph: WorkflowGraphReadonly) {
+    const logicalPlan = ExecuteWorkflowService.getLogicalPlanRequest(workflowGraph);
+    const operators = logicalPlan.operators;
+    const links = logicalPlan.links;
+    const websocketUrl =
+      getWebsocketUrl(WorkflowWebsocketService.TEXERA_WEBSOCKET_ENDPOINT) +
+      (environment.userSystemEnabled && AuthService.getAccessToken() !== null
+        ? "?access-token=" + AuthService.getAccessToken()
+        : "");
+    this.websocket = webSocket<TexeraWebsocketEvent | TexeraWebsocketRequest>(websocketUrl);
+    // setup reconnection logic
+    const wsWithReconnect = this.websocket.pipe(
+      retryWhen(errors =>
+        errors.pipe(
+          tap(_ => (this.isConnected = false)), // update connection status
+          tap(_ =>
+            console.log(`websocket connection lost, reconnecting in ${WS_RECONNECT_INTERVAL_MS / 1000} seconds`)
+          ),
+          delayWhen(_ => timer(WS_RECONNECT_INTERVAL_MS)), // reconnect after delay
+          tap(_ => {
+            this.send("RegisterEIdRequest", { eId, operators, links }); // re-register wid
```

Comment:
update the comment accordingly. `register execution details`

Reply:
done